src/Microsoft.AspNetCore.OData/EnableNestedPathsAttribute.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNet.OData.Adapters;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.AspNet.OData.Query;
+using Microsoft.AspNet.OData.Routing;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// This class defines an attribute that can be applied to an action to allow handling
+    /// of arbitrarily nested paths. The result of the action should be an IQueryable
+    /// or SingleResult. The sequence of property and key accesses in the path will be
+    /// applied to the result of the action through query transformations.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
+    public sealed class EnableNestedPathsAttribute : ActionFilterAttribute
+    {
+        /// <inherit/>
+        public EnableNestedPathsAttribute(): base()
+        {
+            // ensures this filter is executed before [EnableQuery] which has default order 0
+            // since the OnActionExecuted method is executed after the action,
+            // a filter with a lower order will be executed after a filter with a higher order
+            Order = 1;
+        }
+        
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="actionExecutedContext"></param>
+        public override void OnActionExecuted(ActionExecutedContext actionExecutedContext)
+        {
+            if (actionExecutedContext == null)
+            {
+                throw Error.ArgumentNull("actionExecutedContext");
+            }
+
+            HttpRequest request = actionExecutedContext.HttpContext.Request;
+            if (request == null)
```

Comment:
Is this an actual possibility? Can `request` ever be null?

Reply:
Without null check, we MAY break some tests in WebAPI

Reply:
EnableQuery has [similar checks](https://github.com/OData/WebApi/blob/5a5c585c139cddc3c63488f03daeb97a4b1ba868/src/Microsoft.AspNetCore.OData/EnableQueryAttribute.cs#L213)